### ISSUE TITLE
Notification popup in AppAlert mode

### ISSUE THREAD

**Issue by eugenio-caicedo** [OneSignal/OneSignal-Android-SDK#256]
My problem is the following: when the app is open the notification is received by the following popup window

![popup window](https://user-images.githubusercontent.com/12780910/27556747-d9d82c7c-5a84-11e7-8c3c-21d10ef6757a.png)

This is not ok. When the user presses the "Ok" button another popup window is showed (it's part of my application), and it is necessary to override the first pop up by the second popup. I see this configuration for the manifest file, but I am not sure if this help

`<meta-data android:name="com.onesignal.NotificationOpened.DEFAULT" android:value="DISABLE" />`

**Comment by jkasten2** [comment]
@eugenio-caicedo You can call [`setInFocusDisplaying`](https://documentation.onesignal.com/docs/android-native-sdk#section--setinfocusdisplaying-) to make this a notification that shows in the shade or disable notifications from showing when your app is in focus.

**Comment by eugenio-caicedo** [comment]
Thanks, @jkasten2 Ok, I am using:

`.inFocusDisplaying(OneSignal.OSInFocusDisplayOption.InAppAlert)`


The possible values are:

`public enum OSInFocusDisplayOption {
      None, InAppAlert, Notification
   }`

But none resolved the problem.

**Comment by jkasten2** [comment]
@eugenio-caicedo If you set it to `Notification` do you get both the alert and the notification in the shade? Can you make sure your not calling OneSignal init more than once?

**Comment by eugenio-caicedo** [comment]
@jkasten2 If I set it to Notification, I get only the notification, not the popup. Really, I don't need both: alert and notification. Only if the app it's open, I need calling to the OneSignalHandler.notificationOpened function It will show the app popup (This is part of my application, not onesignal).

**Comment by jkasten2** [comment]
@eugenio-caicedo I see, so setting the focus to `None` fixed your problem. Not, none of the options resolved your issue. 😄

**Comment by eugenio-caicedo** [comment]
umm @jkasten2 with the None option I get only the notification when the app is open. And the function is not called.

**Comment by jkasten2** [comment]
@eugenio-caicedo `inFocusDisplaying` doesn't affect displaying if your app is not running or is in the background. The notification show always display in the notification shade when the user is not in your app. If this isn't working then please see the troubleshooting guide below.
https://documentation.onesignal.com/docs/notifications-show-successful-but-are-not-being-shown

Which of the 3 callbacks are not working for you?
[`NotificationOpenedHandler`](https://documentation.onesignal.com/docs/android-native-sdk#section--notificationopenedhandler-), [`NotificationReceivedHandler`](https://documentation.onesignal.com/docs/android-native-sdk#section--notificationreceivedhandler-), or [`NotificationExtenderService`](https://documentation.onesignal.com/docs/android-customizations#section-background-data-and-notification-overriding) 

Thanks.

**Comment by eugenio-caicedo** [comment]
@jkasten2 Ok, I explain again. Currently, when the notification is received, the 'NotificationExtenderService.onNotificationProcessing' function is called, I call the displayNotification function inside of it. If the app is open, the "You've been requested!" popup is displayed, but this is not the popup correct. The popup correctly is 'ClaimedRequestDialogFragment', this is a fragment inside my app. This fragment is called when the function OneSignal.NotificationOpenedHandler.notificationOpened is called.

In summary, the flow is this:

1. The server sends the push notification
2. In the app, the function 'NotificationExtenderService.onNotificationProcessing' is called.
3. Inside this function, the displayNotification function is called.
4. If the app is open, the "You've been requested!" popup is displayed.
5. In step 4, the correct popup should be ClaimedRequestDialogFragment, a fragment inside of my app.

**Comment by jkasten2** [comment]
@eugenio-caicedo That is odd, the `notificationOpened` should not be firing in this case. Can you confirm you are using the latest 3.5.8 SDK version?

If so can you get the full stack trace from your open handler? You can use one of these solutions to do so.
https://stackoverflow.com/questions/1069066/get-current-stack-trace-in-java

**Comment by eugenio-caicedo** [comment]
@jkasten2 Ok, you are right. In this flow, the notificationOpened function should not be called because the notification ("You've been requested!" popup) it's not clicked by the user. This is a [video](https://drive.google.com/file/d/0B3g9m4gzKqSYbWI5b2ZDOVFTYk0/view?usp=sharing) with the example: when the ok button is clicked the notificationOpened function is called. The library version is com.onesignal:OneSignal:[3.5.3,4.0.0)

**Comment by jkasten2** [comment]
@eugenio-caicedo You can update it to `com.onesignal:OneSignal:[3.5.8,4.0.0)` to ensure you are using the latest.

Since you have a `NotificationExtenderService` setup you should set the following on OneSignal init.
```Java
.inFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification)
```
Otherwise an alert will always show from our SDK.

**Comment by eugenio-caicedo** [comment]
@jkasten2 I have updated the library version. The notifications are received in the drawer, but the "You've been requested!" popup is not displayed. I need the popup but with the notification effect, I mean the popup but with the OverrideSettings configuration and the notification sounds. Is it possible this?

**Comment by jkasten2** [comment]
@eugenio-caicedo `OverrideSettings` only applies to notifications shown in the shade. You will need to show your own alert to customize it. This can be done by displaying your alert from the [`NotificationReceivedHandler`](https://documentation.onesignal.com/docs/android-native-sdk#section--notificationreceivedhandler-)

**Comment by eugenio-caicedo** [comment]
@jkasten2 Umm I understand, so in the notificationReceived function, Can I redirect to main activity for showing a fragment?

**Comment by jkasten2** [comment]
Correct, from the close of your alert you can trigger an Activity like normal.

**Comment by eugenio-caicedo** [comment]
@jkasten2 Perfect, thank you very much. I'll try this to see what happens.
